File: PocketEdition/RoleRollsPocketEdition/Archetypes/Archetype.cs
using RoleRollsPocketEdition.Bonuses;
using RoleRollsPocketEdition.Core.Entities;

namespace RoleRollsPocketEdition.Archetypes;

public class Archetype : Entity
{
    public string Name { get; set; }
    public string Description { get; set; }
    public ICollection<Bonus> Bonuses { get; set; }
}

File: frontend/role-rolls/src/app/models/archetypes/archetype.ts
import {Bonus} from '@app/models/bonuses/bonus';
import { Spell } from '@app/models/spells/spell';

export interface Archetype {
  id: string;
  name: string;
  description: string;
  bonuses: Bonus[];
  powerDescriptions: ArchetypePowerDescription[];
  spells?: Spell[];
}
export interface ArchetypePowerDescription {
  id: string;
  description: string;
  name: string;
  gameDescription: string;
  level: number;
}
